pkg/cloudprovider/vsphereparavirtual/controllers/routablepod/core.go

```diff
@@ -16,52 +16,41 @@ package routablepod
 import (
 	"context"
 	"fmt"
-	"time"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
-	ippoolclientset "k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphereparavirtual/client/clientset/versioned"
-	ippoolscheme "k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphereparavirtual/client/clientset/versioned/scheme"
-	ippoolinformers "k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphereparavirtual/client/informers/externalversions"
 	"k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphereparavirtual/controllers/routablepod/ippool"
 	"k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphereparavirtual/controllers/routablepod/node"
+	"k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphereparavirtual/ippoolmanager"
 	k8s "k8s.io/cloud-provider-vsphere/pkg/common/kubernetes"
-)
-
-const (
-	defaultResyncTime time.Duration = time.Minute * 1
+	"k8s.io/klog/v2"
 )
 
 // StartControllers starts ippool_controller and node_controller
-func StartControllers(scCfg *rest.Config, client kubernetes.Interface, informerManager *k8s.InformerManager, clusterName, clusterNS string, ownerRef *metav1.OwnerReference) error {
+func StartControllers(scCfg *rest.Config, client kubernetes.Interface,
+	informerManager *k8s.InformerManager, clusterName, clusterNS string, ownerRef *metav1.OwnerReference, vpcMode bool) error {
+
 	if clusterName == "" {
 		return fmt.Errorf("cluster name can't be empty")
 	}
 	if clusterNS == "" {
 		return fmt.Errorf("cluster namespace can't be empty")
 	}
-	ipcs, err := ippoolclientset.NewForConfig(scCfg)
-	if err != nil {
-		return fmt.Errorf("error building ippool clientset: %w", err)
-	}
 
-	s := scheme.Scheme
-	if err := ippoolscheme.AddToScheme(s); err != nil {
-		return fmt.Errorf("failed to register ippoolSchemes")
-	}
```

Comment:
Manual AddToScheme is removed in new code, because it will be automatically added when we import the client lib. See [this](https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/cloudprovider/vsphereparavirtual/client/clientset/versioned/scheme/register.go#L53-L56) and [this](https://github.com/vmware-tanzu/nsx-operator/blob/vpc_dev/pkg/client/clientset/versioned/scheme/register.go#L42-L45)